pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -118,7 +123,8 @@ let
     (optionalString (isGhcjs || versionOlder "7" ghc.version) (enableFeature enableStaticLibraries "library-vanilla"))
     (optionalString (isGhcjs || versionOlder "7.4" ghc.version) (enableFeature enableSharedExecutables "executable-dynamic"))
     (optionalString (isGhcjs || versionOlder "7" ghc.version) (enableFeature doCheck "tests"))
-  ] ++ optionals (enableDeadCodeElimination && (stdenv.lib.versionOlder "8.0.1" ghc.version)) [
+  ] ++ optional doCoverage "--enable-coverage"
```

Comment:
I would like 

    optionallString (stdenv.lib.versionOlder "7.8" ghc.version) (enableFeature doCheck "coverage")

best. It's longer and more complicated, but I like the notion that features, which are not enabled, are explicitly disabled at configure time.